src/wing/CCPACSWingCell.cpp

```diff
@@ -663,33 +815,139 @@ void CCPACSWingCell::BuildSkinGeometry(GeometryCache& cache) const
     // get the shape of the skin
     TopoDS_Shape loftShape = m_parent->GetParent()->GetLoftSide() == UPPER_SIDE ? wsr.GetUpperShape() : wsr.GetLowerShape();
 
-    // calculate corner points on chord face and project points on loftShape
-    // cache points for later use
-    cache.IBLE = wsr.GetPoint(m_etaXsiCache->innerLeadingEdgePoint.eta,
-                              m_etaXsiCache->innerLeadingEdgePoint.xsi,
-                              WING_COORDINATE_SYSTEM);
-    cache.projectedIBLE = ProjectPointOnShape(loftShape, cache.IBLE, zRefDir);
-    cache.IBTE = wsr.GetPoint(m_etaXsiCache->innerTrailingEdgePoint.eta,
-                              m_etaXsiCache->innerTrailingEdgePoint.xsi,
-                              WING_COORDINATE_SYSTEM);
-    cache.projectedIBTE = ProjectPointOnShape(loftShape, cache.IBTE, zRefDir);
-    cache.OBLE = wsr.GetPoint(m_etaXsiCache->outerLeadingEdgePoint.eta,
-                              m_etaXsiCache->outerLeadingEdgePoint.xsi,
-                              WING_COORDINATE_SYSTEM);
-    cache.projectedOBLE = ProjectPointOnShape(loftShape, cache.OBLE, zRefDir);
-    cache.OBTE = wsr.GetPoint(m_etaXsiCache->outerTrailingEdgePoint.eta,
-                              m_etaXsiCache->outerTrailingEdgePoint.xsi,
-                              WING_COORDINATE_SYSTEM);
-    cache.projectedOBTE = ProjectPointOnShape(loftShape, cache.OBTE, zRefDir);
-
-    // cut the shape at the cell borders
-    TopoDS_Shape resultShape = CutSpanwise(cache, loftShape, SpanWiseBorder::Inner, m_positioningInnerBorder, zRefDir, 1e-2);
-    resultShape              = CutSpanwise(cache, resultShape, SpanWiseBorder::Outer, m_positioningOuterBorder, zRefDir, 1e-2);
-    resultShape              = CutChordwise(cache, resultShape, ChordWiseBorder::LE, m_positioningLeadingEdge, zRefDir, 1e-2);
-    resultShape              = CutChordwise(cache, resultShape, ChordWiseBorder::TE, m_positioningTrailingEdge, zRefDir, 1e-2);
-
-    // store the result
-    cache.skinGeometry = resultShape;
+    int use_contour_coordinates = (int)(m_positioningInnerBorder.GetInputType()  == CCPACSWingCellPositionSpanwise::Contour);
```

Comment:
I don't want to be too picky, but stricktly speaking, the undescore variable naming violates TiGLs code style ;)